### PR TITLE
Plugin doc migration from wiki to github

### PR DESCRIPTION
**Pull request by Mr-DG-Wick**
- Migrated some Doc from https://github.com/jenkins-infra/plugins-wiki-docs/tree/master/kerberos-sso
- Created and updated a CHANGELOG.md file
- Migrated images
- Fixed minor typos and modified grammar.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did

### CODE REVIEW

**Comment by olivergondza** [comment]
Thanks for the worn on unifying this! Please do not remove https://github.com/jenkinsci/kerberos-sso-plugin/blob/b2657c08e49320b570152b53d6eaa0b1b847cf14/README.md?plain=1#L15=L36, it is relevant for the CasC users.

**Comment by Mr-DG-Wick** [comment]
> Thanks for the worn on unifying this! Please do not remove https://github.com/jenkinsci/kerberos-sso-plugin/blob/b2657c08e49320b570152b53d6eaa0b1b847cf14/README.md?plain=1#L15=L36, it is relevant for the CasC users.

Thanks for the info @olivergondza !. I too was skeptical about this. I shall add it.

File: README.md [hunk 1]
```diff
@@ -6,38 +6,177 @@
...
+
+## Setup guide
+
+Since this plugin took us so long to set up correctly, we'll provide a
+short list of steps and hints on how this setup can be done. Some of
+these tips can also be found in the help links in the configuration
+page. This guide is for a server using **Linux**. We have not
+successfully configured the plugin to run on a Windows server. Please
+inform us if you succeed in doing so, or just add the steps here on the
+wiki.
```
Comment (rsandell) [inline README.md:54]: This is no longer on the wiki. But a bit unsure how to rephrase it.

File: README.md [hunk 2]
```diff
@@ -6,38 +6,177 @@
...
+
+-   **The service account** is only used when a keytab is not present on
+    the server. Keytabs are probably used either everywhere or nowhere
+    on the intranet. Although, if there are Service accounts at hand, it
+    can only benefit the stability of the solution to provide it. Some
+    Kerberos information sources, such as Spnego may call the service
+    account "pre-auth username and password"
+
+&nbsp;
```
Comment (rsandell) [inline README.md:63]: Are all these of any use?

**Comment by rsandell** [comment]
I personally prefer user facing docs to be in a separate `docs` directory than the root `README` file. That file imo is for developer docs/ those who are browsing the repository.
But this is Oliver's plugin now, so it's up to him ;)

Also don't forget to update the pom file with the new location, or the move wont have any effect.

**Comment by olivergondza** [comment]
Thanks @Mr-DG-Wick, @StackScribe  and @rsandell. Merged with some massaging!

**Pull request merged**